### PR TITLE
Add RabbitMQ driver & refactoring code

### PR DESCRIPTION
**Pull request by bayram-aloware**
This PR adds RabbitMQ support to this package.
With refactored code we are able add any other driver later if its required.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,34 +18,116 @@ php artisan horizon:wildcard-consume
...
-                'queue' => ['c_*', 'company_*_messages'],
-                'balance' => 'auto',
-                'processes' => 5,
-                'tries' => 3,
-            ],
+'environments' => [
+    'production' => [
+        'supervisor-1' => [
+            'connection' => 'redis',
+            'queue' => ['default', 'a_*', 'user_*_notes', '*-bills'],
```
Comment (mohsen-aloware) [inline README.md:32]: let's remove that meaningless a_*

File: README.md [hunk 2]
```diff
@@ -19,34 +18,116 @@ php artisan horizon:wildcard-consume
...
+            'connection' => 'redis',
+            'queue' => ['c_*', 'company_*_messages'],
+            'balance' => 'auto',
+            'processes' => 5,
+            'tries' => 3,
+        ],
+    ],
+]
 ```
+
```
Comment (mohsen-aloware) [inline README.md:54]: just mention about the dependency and reference to the main repository for more information on their config file.

**Review by mohsen-aloware — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -19,34 +18,116 @@ php artisan horizon:wildcard-consume
...
+### Example config for RabbitMQ
```
Comment (mohsen-aloware) [inline README.md:55]: just mention about the dependency and reference to the main repository for more information on their config file.

Comment (bayram-aloware) [reply README.md:55]: OK

**Pull request merged**